Scripts/BestiaryMain.cs

```diff
@@ -434,35 +469,51 @@ public object NewSaveData()
                 ConvertedBestiarySaveData = true,
             };
         }
+
         public object GetSaveData()
         {
             return new MyModSaveData
             {
-                ConvertedBestiarySaveData = ConvertedBestiarySaveData,
+                ConvertedBestiarySaveData = true,
                 KillCounts = killCounts,
                 UnlockedBestiary = UnlockedBestiary
             };
         }
+
         public void RestoreSaveData(object saveData)
         {
             var myModSaveData = (MyModSaveData)saveData;
 
             if (!myModSaveData.ConvertedBestiarySaveData)
             {
-                var newKillCounts = new Dictionary<string, uint>();
-
-                foreach (var KillCount in myModSaveData.KillCounts)
+                try
                 {
-                    newKillCounts.Add(BestiarySaveConvert.Convert(KillCount.Key), KillCount.Value);
-                }
+                    var newKillCounts = new Dictionary<string, uint>();
+
+                    foreach (var KillCount in myModSaveData.KillCounts)
+                    {
+                        newKillCounts.Add(BestiarySaveConvert.Convert(KillCount.Key), KillCount.Value);
+                    }
 
-                myModSaveData.KillCounts = newKillCounts;
-                myModSaveData.ConvertedBestiarySaveData = true;
+                    myModSaveData.KillCounts = newKillCounts;
+                }
+                catch
```

Comment:
If I'm reading this correctly, this catches all thrown exceptions, which isn't ideal.
[Some](https://stackoverflow.com/questions/12911833/how-to-catch-all-exceptions-in-c-sharp-using-try-and-catch) [info](https://stackoverflow.com/questions/11191260/should-you-catch-all-exceptions)

Reply:
Here I wanted to protect my save conversion.
I will add `(Exception ex)` with some logging if it is preferable. 
Other than that, it is ready. Though I have a question about `GetLuckRoll` function. Do you think it will work? I added more randomize to that.

Reply:
Added exception log. If you're good with `GetLuckRoll`, then I have nothing to add so far.

Reply:
> Added exception log. If you're good with `GetLuckRoll`, then I have nothing to add so far.

Nice

> Other than that, it is ready. Though I have a question about GetLuckRoll function. Do you think it will work? I added more randomize to that.

It seems good to me. Here is how Magicono [implemented](https://github.com/magicono43/DFU-Mod_Jewelry-Additions/blob/2faf117b9912e1848e07e470b940b694dd7791f4/Scripts/JewelryAdditionsMain.cs#L367C20-L367C20) a similar drop check for reference, but yours seems good too.

```
// Approx. 1 in 120 enemies
return dropChance > 0.8;
```
If you want to, you could make `0.8` a settings option to appease anyone who is unhappy with the drop chances.

Great work!

Reply:
> If you want to, you could make 0.8 a settings option to appease anyone who is unhappy with the drop chances.

Awesome idea! I'm on it! )))

Reply:
@Shapur1234

Reply:
Last question, before we merge. If you like and agree on proposed DREAM patch look, I'll build it and add into PR 

If not - then you can merge 😁